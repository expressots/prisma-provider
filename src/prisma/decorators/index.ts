export { prismaField, Default } from "./prisma-field.decorator";
export { prismaIndex } from "./prisma-index.decorator";
export { prismaModel } from "./prisma-model.decorator";
export { prismaRelation } from "./prisma-relation.decorator";
