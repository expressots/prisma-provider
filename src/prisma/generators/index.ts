export { codeFirstGen } from "./model-generator";
export { removeModels } from "./model-remover";
