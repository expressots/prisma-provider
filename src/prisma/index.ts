export * from "./decorators";
export * from "./generators";
