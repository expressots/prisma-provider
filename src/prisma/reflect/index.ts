export { reflect } from "./reflect";
export { processFile } from "./process-file";
export { FileArray, FileGlob, FileModule } from "./file-module";
export { FileInfo, PropertyInfo } from "./file-info";
export * from "./extractor";
