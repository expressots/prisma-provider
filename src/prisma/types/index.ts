export * from "./prisma.types";
export * from "./scalar.types";
export * from "./typeAttributes";