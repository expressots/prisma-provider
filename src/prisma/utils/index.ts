export { ModelsOf } from "./model-prisma-client.extractor";
