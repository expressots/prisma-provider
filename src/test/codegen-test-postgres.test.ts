import fs from "fs";
import { codeFirstGen } from "../prisma";

describe("Create expressots.config.ts", () => {
    const configFileContent = `import { ExpressoConfig, Pattern } from "./src/@types/config";
const config: ExpressoConfig = {
    sourceRoot: "src",
    scaffoldPattern: Pattern.KEBAB_CASE,
    opinionated: true,
    providers: {
        prisma: {
            schemaName: "schema",
            schemaPath: "src/test",
            entitiesPath: "test/entities-postgres",
            entityNamePattern: "entity",
        },
    },
};

export default config;
`;

    const schemaFileContent = `generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgres"
  url      = "test"
}

enum UnusedEnum {
  TEST1
}
`;

    beforeAll(() => {
        const filePath = "expressots.config.ts";
        fs.writeFileSync(filePath, configFileContent, { encoding: "utf8" });

        const filePathSchemaPrisma = "./src/test/schema.prisma";
        fs.writeFileSync(filePathSchemaPrisma, schemaFileContent, { encoding: "utf8" });
    });

    afterAll(() => {
        const filePath = "expressots.config.ts";
        fs.unlinkSync(filePath);

        const filePathSchemaPrisma = "./src/test/schema.prisma";
        fs.unlinkSync(filePathSchemaPrisma);
    });

    it("should create expressots.config.ts file with the correct content", () => {
        const filePath = "expressots.config.ts";
        expect(fs.existsSync(filePath)).toBe(true);

        const fileContent = fs.readFileSync(filePath, { encoding: "utf8" });
        expect(fileContent).toMatchSnapshot();
    });

    it("Check schema.prisma", async () => {
        const filePathSchemaPrisma = "./src/test/schema.prisma";
        expect(fs.existsSync(filePathSchemaPrisma)).toBe(true);

        await codeFirstGen();

        const fileContent = fs.readFileSync(filePathSchemaPrisma, { encoding: "utf8" });
        expect(fileContent).toMatchSnapshot();
    }, 120000);
});
