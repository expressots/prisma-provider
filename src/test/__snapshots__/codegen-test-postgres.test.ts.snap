// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create expressots.config.ts Check schema.prisma 1`] = `
"generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgres"
  url      = "test"
}

model CategoriesOnPosts {
  id         Int       @id @default(autoincrement())
  asignedAt  DateTime  @default(now())
  assignedBy String
  category   Category? @relation(fields: [categoryId], references: [id])
  categoryId String?   @unique
  post       Post?     @relation(fields: [postId], references: [id])
  postId     String?   @unique
}

model Category {
  id    String              @id @default(uuid())
  name  String              @default(uuid())
  posts CategoriesOnPosts[]
  post  Post[]

  @@index([name], name: "name")
}

model Post {
  id          String              @id
  title       String
  content     String
  categories  Category[]
  categories2 CategoriesOnPosts[]
  user        User?               @relation(fields: [userId], references: [id])
  userId      String?             @unique
}

model Profile {
  id      String  @id
  content String
  user    User?   @relation(fields: [userId], references: [id], onDelete: Cascade, onUpdate: Cascade)
  userId  String? @unique
}

model User {
  id      String   @id
  email   String   @unique
  role    Role
  profile Profile?
  posts   Post[]
}

enum Role {
  ADMIN
  REGULAR
}
"
`;

exports[`Create expressots.config.ts should create expressots.config.ts file with the correct content 1`] = `
"import { ExpressoConfig, Pattern } from "./src/@types/config";
const config: ExpressoConfig = {
    sourceRoot: "src",
    scaffoldPattern: Pattern.KEBAB_CASE,
    opinionated: true,
    providers: {
        prisma: {
            schemaName: "schema",
            schemaPath: "src/test",
            entitiesPath: "test/entities-postgres",
            entityNamePattern: "entity",
        },
    },
};

export default config;
"
`;
